modules/vnet/main.tf

```diff
@@ -20,14 +20,26 @@ locals {
 }
 
 resource "azurerm_subnet" "this" {
-  for_each = var.subnets
+  for_each = var.create_subnets ? var.subnets : {}
 
   name                 = each.key
   resource_group_name  = var.resource_group_name
   virtual_network_name = local.virtual_network.name
   address_prefixes     = each.value.address_prefixes
 }
 
+data "azurerm_subnet" "this" {
+  for_each = var.create_subnets == false ? var.subnets : {}
```

Comment:
for whatever reason this throws an error when used in glue code (not observed during tests though):
```
Error: Inconsistent conditional result types
│
│   on .terraform/modules/vnet/modules/vnet/main.tf line 23, in resource "azurerm_subnet" "this":
│   23:   for_each = var.create_subnets ? var.subnets : {}
│     ├────────────────
│     │ var.create_subnets is true
│     │ var.subnets is object with 3 attributes
│
│ The true and false result expressions must have consistent types. The 'true' value includes object attribute "management", which is absent in the 'false' value.
```